vts/trustedservices/trustedservices_grpc.go

```diff
@@ -360,38 +360,44 @@ func (o *GRPC) GetAttestation(
 		return o.finalize(appraisal, err)
 	}
 
-	appraisal.EvidenceContext.ReferenceId = extracted.ReferenceID
+	appraisal.EvidenceContext.ReferenceIds = extracted.ReferenceIDs
 
 	o.logger.Debugw("constructed evidence context",
-		"software-id", appraisal.EvidenceContext.ReferenceId,
-		"trust-anchor-id", appraisal.EvidenceContext.TrustAnchorId)
+		"software-id", appraisal.EvidenceContext.ReferenceIds,
+		"trust-anchor-id", appraisal.EvidenceContext.TrustAnchorIds)
 
-	endorsements, err := o.EnStore.Get(appraisal.EvidenceContext.ReferenceId)
-	if err != nil && !errors.Is(err, kvstore.ErrKeyNotFound) {
-		return o.finalize(appraisal, err)
-	}
+	var tendorsements []string
+	for _, reference := range appraisal.EvidenceContext.ReferenceIds {
+
+		endorsements, err := o.EnStore.Get(reference)
+		if err != nil && !errors.Is(err, kvstore.ErrKeyNotFound) {
+			return o.finalize(appraisal, err)
+		}
+
+		if len(endorsements) > 0 {
+			o.logger.Debugw("obtained endorsements", "endorsements", endorsements)
+		}
+		tendorsements = append(tendorsements, endorsements...)
```

Comment:
Just a thought.  Why make it conditional?  It'd be also interesting to intercept the complementary condition, i.e., when an empty endorsement is returned, and maybe flag the anomaly.  In fact, I'm not sure we allow the returned ref-values to be empty...

Reply:
Returned Reference Values as Empty is a valid condition, it is up to the `scheme plugin` to flag it as an Error: if a specific implementation mandates a requirement  for a non-zero reference values!

Reply:
I will add a Debug to print the anomaly any way!

Reply:
> Returned Reference Values as Empty is a valid condition, it is up to the `scheme plugin` to flag it as an Error: if a specific implementation mandates a requirement for a non-zero reference values!

hmm, I am not sure about this.  Before this change, an "empty" endorsement would be signalled by returning `kvstore.ErrKeyNotFound`

Reply:
The existing `kvstore.ErrKeyNotFound` change still exists!

This should be the case, where the `key` exist in the Store but no associated `vals` detected for the specific `key`

Reply:
> This should be the case, where the `key` exist in the Store but no associated `vals` detected for the specific `key`

How can this happen?

Reply:
Checked the `kvstore` interface. Yes, this is not possible!

Hence removed the top level check for endorsements as this is superficial.